File: dart/com/spoonacular/client/model/inline_response20028_nutrition.dart
part of openapi.api;

class InlineResponse20028Nutrition {
  
  List<RecipesParseIngredientsNutritionNutrients> nutrients = [];
  
  RecipesParseIngredientsNutritionCaloricBreakdown caloricBreakdown = null;
  InlineResponse20028Nutrition();

  @override
  String toString() {
    return 'InlineResponse20028Nutrition[nutrients=$nutrients, caloricBreakdown=$caloricBreakdown, ]';
  }

  InlineResponse20028Nutrition.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['nutrients'] == null) {
      nutrients = null;
    } else {
      nutrients = RecipesParseIngredientsNutritionNutrients.listFromJson(json['nutrients']);
    }
    if (json['caloricBreakdown'] == null) {
      caloricBreakdown = null;
    } else {
      caloricBreakdown = new RecipesParseIngredientsNutritionCaloricBreakdown.fromJson(json['caloricBreakdown']);
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (nutrients != null)
      json['nutrients'] = nutrients;
    if (caloricBreakdown != null)
      json['caloricBreakdown'] = caloricBreakdown;
    return json;
  }

  static List<InlineResponse20028Nutrition> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20028Nutrition>() : json.map((value) => new InlineResponse20028Nutrition.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20028Nutrition> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20028Nutrition>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20028Nutrition.fromJson(value));
    }
    return map;
  }
}

File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20028Nutrition.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.RecipesParseIngredientsNutritionCaloricBreakdown
import com.spoonacular.client.model.RecipesParseIngredientsNutritionNutrients
import scala.collection.immutable.Seq

/**
 * 
 * @param nutrients 
 * @param caloricBreakdown 
 */
case class InlineResponse20028Nutrition(nutrients: Seq[RecipesParseIngredientsNutritionNutrients],
                caloricBreakdown: RecipesParseIngredientsNutritionCaloricBreakdown
                )

object InlineResponse20028Nutrition {
    /**
     * Creates the codec for converting InlineResponse20028Nutrition from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20028Nutrition] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20028Nutrition] = deriveEncoder
}
